src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -124,7 +124,11 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
 
             var provenHeadersPayload = new ProvenHeadersPayload();
 
-            ChainedHeader header = this.FindHeaderToStartFrom(fork);
+            ChainedHeader header = null;
+            if (getHeadersPayload.HashStop != null && getHeadersPayload.HashStop != uint256.Zero)
+                header = this.FindHeaderToStartFromByHashStop(getHeadersPayload.HashStop);
```

Comment:
The content of `FindHeaderToStartFromByHashStop` and `FindHeaderToStartFrom` are practically identical you could easily just call `FindAncestorOrSelf` at the caller if hashstop is not null and just have one method

Reply:
This is also not so correct because if hashstop is has a value the fork needs to change to the value in hashstop

Reply:
Ah sorry no this is correct however we need to make sure hashstop is between fork and starting header